### PR TITLE
Http timeouts/retries

### PR DESCRIPTION
**Pull request by cavis**

### CODE REVIEW

File: lib/proxy.js [hunk 1]
```diff
@@ -15,13 +15,19 @@ module.exports = class Proxy {
     this.protocol = useHttps ? https : http;
   }
 
+  retries(index) {
+    return [1000, 1000, 5000][index];
```
Comment (cavis) [inline lib/proxy.js]: Not entirely sure what good values here would be.

Comment (kookster) [reply lib/proxy.js]: As we are seeing about a 20% fail rate, I might add one more retry at the 1 sec level

File: lib/proxy.js [hunk 2]
```diff
@@ -53,19 +59,39 @@ module.exports = class Proxy {
...
+
+    // depending on which attempt this is, set a timeout
+    const wait = this.retries(attemptNum);
+    if (wait) {
+      timeout = setTimeout(() => {
+        aborted = true;
+        req.abort();
+
+        const info = `${opts.method} ${opts.hostname}${opts.path}`;
+        console.warn(`[WARN] Timed out after ${wait}ms - ${info}`);
```
Comment (kookster) [inline lib/proxy.js]: nice to have this in the logs

File: index.js [hunk 3]
```diff
@@ -56,10 +56,10 @@ exports.handler = function handler(event, context, callback) {
     }).catch(err => {
       console.error(`[ERROR] 500 ${event.httpMethod} ${event.path}`);
       console.error(err);
-      callback(null, ApiResponse.error('Something went wrong'));
+      callback(null, {statusCode: 500, body: 'Something went wrong', headers: {'content-type': 'text/plain'}});
     });
   } else {
     console.error(`[ERROR] No handler for ${event.httpMethod} ${event.path}`);
-    callback(null, ApiResponse.error('Something went wrong'));
+    callback(null, {statusCode: 500, body: 'Something went wrong', headers: {'content-type': 'text/plain'}});
```
Comment (kookster) [inline index.js]: do you want a different body message in this case, to differentiate?

**Pull request merged**